crates/ty_python_semantic/resources/mdtest/attributes.md

```diff
@@ -1721,7 +1721,7 @@ d = True
 reveal_type(d.__class__)  # revealed: <class 'bool'>
 
 e = (42, 42)
-reveal_type(e.__class__)  # revealed: <class 'tuple'>
+reveal_type(e.__class__)  # revealed: <class 'tuple[Literal[42], ...]'>
```

Comment:
what do we think the type of `e.__class__()` is?

For this snippet:

```py
def f(x: tuple[int, int]):
    reveal_type(x.__class__())
```

Pyright incorrectly says `tuple[int, int]`.

Mypy does something more defensible here: it says:

```
main.py:2: error: Cannot instantiate type "Type[tuple[int, int]]"  [misc]
```

and reveals `Any`.

Reply:
We now reveal this as `tuple[Literal[42], Literal[42]]`, tracking the full detail of the tuple elements even in the generic alias that we create for the tuple's class